### PR TITLE
[#26] [Backend] As a user, I can see my user profile on Home screen

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/luongvo/flutter-survey/issues/26

## What happened 👀

After logged in successfully, I can see my user profile on `Home` screen.

- Call `GET {{base-url}}/api/v1/me` with Authorization.
 
## Insight 📝

- We need to setup a new `UserService`, `UserRepository` to include new APIs about user.
 
## Proof Of Work 📹

```
I/flutter (29682): *** Request ***
I/flutter (29682): uri: https://survey-api.nimblehq.co/api/v1/surveys?page%5Bnumber%5D=1&page%5Bsize%5D=10
I/flutter (29682): method: GET
I/flutter (29682): responseType: ResponseType.json
I/flutter (29682): followRedirects: true
I/flutter (29682): connectTimeout: 3000
I/flutter (29682): sendTimeout: 0
I/flutter (29682): receiveTimeout: 5000
I/flutter (29682): receiveDataWhenStatusError: true
I/flutter (29682): extra: {}
I/flutter (29682): headers:
I/flutter (29682):  Authorization: Bearer AIiRU0SV37LK16YmJAmN-LeClKWZjD_4gwJCIkC4fdc
I/flutter (29682): data:
I/flutter (29682): {}
I/flutter (29682): 
V/FA      (29682): Inactivity, disconnecting from the service
I/flutter (29682): *** Response ***
I/flutter (29682): uri: https://survey-api.nimblehq.co/api/v1/me
I/flutter (29682): statusCode: 200
I/flutter (29682): headers:
I/flutter (29682):  date: Mon, 18 Jul 2022 06:54:18 GMT
I/flutter (29682):  transfer-encoding: chunked
I/flutter (29682):  content-encoding: gzip
I/flutter (29682):  vary: Accept-Encoding, Origin
I/flutter (29682):  server: cloudflare
I/flutter (29682):  x-request-id: 9d3c1460-aec4-4ea1-9858-33d3f2195a38
I/flutter (29682):  cf-ray: 72c94e203cfe221c-HKG
I/flutter (29682):  etag: W/"7b5ecbae2e28b0fa6089654477f8548e"
I/flutter (29682):  x-frame-options: SAMEORIGIN
I/flutter (29682):  connection: keep-alive
I/flutter (29682):  cache-control: max-age=0, private, must-revalidate
I/flutter (29682):  strict-transport-security: max-age=31536000; includeSubDomains
I/flutter (29682):  referrer-policy: strict-origin-when-cross-origin
I/flutter (29682):  cf-cache-status: DYNAMIC
I/flutter (29682):  report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=q%2BnBpQusNx5%2BqXQm3500f%2FB2ifW8FW4nfmpdtRr846r3sKTVj2PLkgdetm0I9MH38yglNZPfjaHgs58H32YmBfHWXlBo7OmIvm4%2BGg0NxUnTVxNzo09gltcNsa7Z20o%2Bfxyszx%2BuLkPdSAKzwm2c6XotmTco"}],"group":"cf-nel","max_age":604800}
I/flutter (29682):  x-permitted-cross-domain-policies: none
I/flutter (29682):  content-type: application/json; charset=utf-8
I/flutter (29682):  expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
I/flutter (29682):  x-xss-protection: 1; mode=block
I/flutter (29682):  alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
I/flutter (29682):  x-download-options: noopen
I/flutter (29682):  nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
I/flutter (29682):  x-runtime: 0.050944
I/flutter (29682):  via: 1.1 vegur
I/flutter (29682):  x-content-type-options: nosniff
I/flutter (29682): Response Text:
I/flutter (29682): {"data":{"id":"31","type":"user","attributes":{"email":"luong@nimblehq.co","name":"Luong","avatar_url":"https://secure.gravatar.com/avatar/8fae17b9d0c4cca18a9661bcdf650f23"}}}
```

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
Nice works

File: lib/api/user_service.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_survey/api/response/user_response.dart';
+import 'package:retrofit/retrofit.dart';
+
+part 'user_service.g.dart';
+
+class SurveyApi {
+  SurveyApi._();
+}
```
Comment (manh-t) [inline lib/api/user_service.dart:9]: is this class redundant? I see it's unused 💭

**Review by manh-t — COMMENTED** [review]
rest LGTM

File: lib/api/user_service.dart (hunk 1 above)
Comment (luongvo) [reply lib/api/user_service.dart]: oops, I was copied it from `survey_service`. Fixed https://github.com/luongvo/flutter-survey/pull/69/commits/1b0740e2062897e2cef055213be24ca43b84114c 😂

**Review by manh-t — APPROVED** [review]
LGTM 🪨

**Pull request merged**